ofrak_core/ofrak/core/openwrt.py

```diff
@@ -433,10 +436,17 @@ async def pack(self, resource: Resource, config=None):
         repacked_data_l.append(await kernel.resource.get_data())
 
         trx_rootfs_offset = trx_kernel_offset + len(repacked_data_l[-1])
-        repacked_data_l.append(await rootfs.resource.get_data())
+        rootfs_data = await rootfs.resource.get_data()
+        if len(rootfs_data) > 0:
+            repacked_data_l.append(rootfs_data)
+        else:
+            trx_rootfs_offset = 0
 
         if header.get_version() == OpenWrtTrxVersion.VERSION2:
-            trx_binheader_offset = trx_rootfs_offset + len(repacked_data_l[-1])
+            if trx_rootfs_offset > 0:
+                trx_binheader_offset = trx_rootfs_offset + len(repacked_data_l[-1])
```

Comment:
Are these supposed to be 2 different cases and there's a typo, or should they be merged into one?

Reply:
I believe that these do need to be two separate cases - `trx_binheader_offset` should always be set to the previous offset + the size of the previous data segment, but this could be either the rootfs or the kernel depending on whether or not the rootfs is included.